anti_patterns/catch_panic.md

```diff
@@ -0,0 +1,79 @@
+# catch_unwind for exceptions
+
+## Description
+
+This anti-pattern arises when the method for catching ([panic::catch_unwind](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html)) an
+unexpected problem (implementation bug) is used to handle an expected problem,
+such as a missing file.
+
+## Example
+
+```rust
+use std::io::prelude::*;
+use std::fs::File;
+use std::panic;
+
+fn main() {
+    // panic::catch_unwind catches any panic that occurs within the
+    // function passed to it
+    let result = panic::catch_unwind(|| {
+        let mut file_result = File::open("foo.txt");
+        file_result.unwrap(); // causes a panic if the file is not found
+    });
+
+    // the program continues running despite the panic
+    println!("potential error: {:?}", result);
+    assert!(result.is_err());
+}
+```
+
+## Motivation
+
+In rust, there are two ways an operation can fail: An expected problem, like a
+file not being found, or a HTTP request timing out. Or, an unexpected problem,
+such as an index being out of bounds, or an assert!() failing. These are
+unexpected because they are bugs that should not happen. It would not make sense
+to handle an error for QuickSort returning an incorrectly unsorted array, as
+this should not happen.
+
+## Advantages
+
+There are scenarios where using `panic::catch_unwind` is the correct choice, e.g. a
```

Comment:
```suggestion
When code is split between multiple threads, or wants to create an isolation boundary, `panic::catch_unwind` is the correct choice. For example, a
```

Reply:
I don't understand "wants to create an isolation boundary". But good to have solid examples.

Reply:
I thought the phrase "isolation boundary" appeared in the nomicon, but looking for it again, I misremembered. I was thinking of something like exception safety [similar to this example](https://doc.rust-lang.org/nomicon/exception-safety.html#vecpush_all), but guarding against "untrusted" code rather than "unsafe" code.

In addition to perhaps threading, that was what I imagined the Web Server example was doing.